### PR TITLE
Pull Request to the code review

### PR DESCRIPTION
**Pull request by cgcarlosg**
This is a Project that uses Ruby on Rails a site similar to private Eventbrite which allows users to create events and then manage user signups. In this app, you can create an event, "attend" your event while being signed in by simply using your username.

### CODE REVIEW

**Comment by kevcap** [comment]
Hey team, how's it going?

Please submit again so other TSE can review it. I wasn't able review your code due issues while trying to run the code in my machine.
P.S.: I marked it as invalid just for bureaucratic questions. You can just submit it right away.

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

However, in order to accomplish the actual page, you required to do these changes, listed below:
### General requirements
- Please next time be careful about commit messages. Add more descriptive commit messages adding what and why you made those changes. And avoid repeating the same message many times, on each commit you're making different changes, write those changes 
image
### SECTION:  event show
- On the event's show page display the list of attendants ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/events_attendees_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
### FILE: app/controllers/events_controller.rb 
* [x] Add this variable past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] Add this variable  ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
  - When you have two lists it's ok, however, when this application will grow that list it's going to be uncountable, that's why we need to add in separate variables 
### FILE: app/controllers/users_controller.rb 
* [x]  Add this variable events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x]  Add this variable events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] Add this variableupcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
  - When you have two lists it's ok, however, when this application will grow that list it's going to be uncountable, that's why we need to add in separate variables 
### Routes file
- Use only for used actions, i.e. `resources :events, only: [:new, :create, :show, :index]`

### FILE: README.md
- Add styling to this readme file (links, code, etc), follow this [tutorial](https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/basic-writing-and-formatting-syntax). Because just plain text is not attractive to read, but with great styling, it makes it more interesting 
- Acknowledgement section. Add links referring to each line. Because when someone ho wnt to check this project, they got in the acknowledgment section, and they want to know for example for about Active Record Queries and Associations., it'd be great to have a direct link to that page  

### OPTIONAL
- If you feel like you really want to go beyond requirements and learn more about associations, work on these ones
* [ ] route under `/event/:event_id/invites/new`
* [ ] Invite model `belongs_to :event`
* [ ] Invite model `belongs_to :user`
* [ ] User model `has_many :invitations`
* [ ] Event model `has_many :invitations`
* [ ] Join model (EventAttendance) is required to make sure that attendance is not created without invitation (e.g. `before_create` callback usage)
### Summary

-  Work on `Routes` file
- It's missing to add some variables in some controllers 

***

Guys work on the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  has_many :events, inverse_of: :creator
+  has_many :enrollments
+  has_many :attended_events, through: :enrollments, source: :event
+
+  scope :past, -> { where('date < ?', Date.today) }
+  scope :upcoming, -> { where('date >= ?', Date.today) }
```
Comment (rindrajosia) [inline app/models/user.rb:8]: We don't need this for the user model. We don't have a date filed for a user. We only needed this for the event model.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
**Changes Required**
Hi there team!
Great job so far
But before moving on there are some changes that we need to work on.

- [ ] [REQUIRED]**The user's show page:**displays events **the user has created**(only)  and events that **the user is attending**(only)  to separated in past and upcoming. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [ ] [REQUIRED] Navigation links are present at the top of the page  (These links should be displayed once the user is logged in: the event's index page, the event's show page, The user's show page).

- Let's include the following gem in our Gemfile: 
```
group :test do
  gem 'database_cleaner-active_record'
end
```

After working on the changes, submit another request for review.

Happy Coding
Rindra Josia

File: app/models/user.rb (hunk 1 above)
Comment (cgcarlosg) [reply app/models/user.rb:8]: Hi.  this requirement says:

- [ ] [REQUIRED] Navigation links are present at the top of the page  (These links should be displayed once the user is logged in: the event's index page, the event's show page, The user's show page).

We include the screenshot, you can check the navbar with the links, is that what you are saying? sorry but that is what we understand.
 
image

Comment (rindrajosia) [reply app/models/user.rb:8]: Hello team!!  Thanks for your reply. However, if you read my comment carefully, I mentioned that: event's index page, event's show page and user's show page should be also in the navigation bar.

**Review by cristianCeamatu — APPROVED** [review]
### 🥇 APPROVED 🥇 Congratulations and good luck with the next projects!

If you ever need any help or notice that you are wasting too many hours on a bug don't hesitate to contact me on Slack.

Best regards,
Cristian Ceamatu
Slack: Cristian Viorel Ceamatu

**Pull request merged**